src/devices/Ads1115/README.md

```diff
@@ -9,10 +9,7 @@ ADS1115 is an Analog-to-Digital converter (ADC) with 16 bits of resolution.
 // set I2C bus ID: 1
 // ADS1115 Addr Pin connect to GND
 I2cConnectionSettings settings = new I2cConnectionSettings(1, (int)I2cAddress.GND);
-// get I2cDevice (in Linux)
-UnixI2cDevice device = new UnixI2cDevice(settings);
-// get I2cDevice (in Win10)
-//Windows10I2cDevice device = new Windows10I2cDevice(settings);
+I2cDevice device = I2cDevice.Create(settings);
```

Comment:
Just as a comment, now that we have I2cDevice.Create method, we might want to think again how most bindings take a I2cDevice as part of their constructor, such that insted they would just create a new one inside their class. There were only two reasons to have it passed in: support for windows/unix (which we no longer need) and in order to allow passing in a device that is not a Unix or Windows I2c device, but instead something that implementes the II2cDevice interface, like a I2C extender.

Reply:
Could the guidance be to pass in I2cDevice? or SpiDevice? similar to IGpioController.  As you mentioned, I could see other flavors (i. e. GpioI2cDevice, GpioSpiDevice) in the future. 

Inside the binding constructor, have something like...

```csharp
_i2cDevice = I2cDevice ?? I2cDevice.Create(settings);
_spiDevice = SpiDevice ?? SpiDevice.Create(settings);
```

Also, with guidance showing the nullable types to pass... could hopefully trigger binding devs to code the binding API with the new C# 8 feature.

Reply:
I'd refrain from that for now unless it covers 95% of the cases. Example thing what I did for SenseHat:
- Lsm9Ds1AccelerometerAndGyroscope doesn't know anything about I2C addresses
- SenseHatAccelerometerAndGyroscope thin wrapper on top of Lsm9Ds1AccelerometerAndGyroscope which only defaults I2C address since it's constant in context of SenseHat

Ideally I think we should create an issue and set up guidelines for this but I think this is non-issue at this point and regardless of this issue we need at least one constructor which takes already existing I2cDevice object.

I don't personally like default value as the ownership is not immediately obvious. Having two classes one which always default and one which never does makes stuff much clearer

Reply:
I'd prefer if we made this non-nullable and create second implementation like DefaultXyzDevice which just inherits, creates I2cDevice and disposes of it. This way you got covered 90% of the cases and advanced usage